File: C#/SleepSort.cs
//  Sleep_Sort Algorithm.
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading;
 
class Program
{
    static void ThreadStart(object item)
    {
        Thread.Sleep(1000 * (int)item);
        Console.WriteLine(item);
    }
 
    static void SleepSort(IEnumerable<int> items)
    {
        foreach (var item in items)
        {
            new Thread(ThreadStart).Start(item);
        }
    }
 
    static void Main(string[] arguments)
    {
        SleepSort(arguments.Select(int.Parse));
    }
}

/* for input 11,6,2,90,23
 * Output is 2 6 11 23 90
 * As input is dynamic so  complexity for can be different
 * Complexity for the sleepSort method : O(NlogN + max(input))
 * Space complexity and time complexity both  are calculated as the function of input size(n).
 */

File: JavaScript/Sleep_Sort.js
function sleepSort(array) {
    array.forEach(e => {
            setTimeout( () => {console.log(e)},e);
    });
}

/* Example
 * Input:
 *    Unsorted Array: Array - e.g.: [ 5,4,7,3,1,8,2,6 ]
 *
 * Output:
 *    1,2,3,4,5,6,7,8
 *
 * Time Complexity: O(max(input))
 * Space Complexity: O(1)
*/
